resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -306,9 +318,14 @@ private[yarn] class YarnAllocator(
       s"executorsStarting: ${numExecutorsStarting.get}")
 
     if (missing > 0) {
-      logInfo(s"Will request $missing executor container(s), each with " +
-        s"${resource.getVirtualCores} core(s) and " +
-        s"${resource.getMemory} MB memory (including $memoryOverhead MB of overhead)")
+      var requestContainerMessage = s"Will request $missing executor container(s), each with " +
+          s"${resource.getVirtualCores} core(s) and " +
+          s"${resource.getMemory} MB memory (including $memoryOverhead MB of overhead) "
```

Comment:
Nitpick: I would remove the trailing space here (so it is exactly the same as before for users), and put it at the beginning of s"and with custom resources:"

Reply:
Fixed